tests/modules/hooks.c

```diff
@@ -266,39 +266,76 @@ void swapDbCallback(RedisModuleCtx *ctx, RedisModuleEvent e, uint64_t sub, void
 /* This function must be present on each Redis module. It is used in order to
  * register the commands into the Redis server. */
 int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+#define VerifySubEventSupported(e, s) \
+    if (!RedisModule_IsSubEventSupported(e, s)) { \
+        return REDISMODULE_ERR; \
+    }
+
     REDISMODULE_NOT_USED(argv);
     REDISMODULE_NOT_USED(argc);
 
     if (RedisModule_Init(ctx,"testhook",1,REDISMODULE_APIVER_1)
         == REDISMODULE_ERR) return REDISMODULE_ERR;
 
     /* replication related hooks */
+    VerifySubEventSupported(RedisModuleEvent_ReplicationRoleChanged, REDISMODULE_EVENT_REPLROLECHANGED_NOW_MASTER);
+    VerifySubEventSupported(RedisModuleEvent_ReplicationRoleChanged, REDISMODULE_EVENT_REPLROLECHANGED_NOW_REPLICA);
```

Comment:
i think it's enough to call this just once to prove it's working, no need to mess up the readability of the test by checking all of them.

Reply:
Wanted to cover as much as possible, but I can remove it.

Reply:
i'm in favor, these tests may serve as both an example and tests + coverage.
for the coverage i think one check is enough. for the example it's somewhat hard to read.

i think you should add just one test, with a comment saying that it's just there as an example or basic coverage.